### PR TITLE
settings fragment

### PR DESCRIPTION
**Pull request by UgurDenli**

### CODE REVIEW

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                Intent.ACTION_PICK,
+                MediaStore.Images.Media.INTERNAL_CONTENT_URI
+            )
+            val ACTIVITY_SELECT_IMAGE = 1234
+            startActivityForResult(i, ACTIVITY_SELECT_IMAGE)
+        }
+
+        binding.accountTransfer.setOnClickListener {
+            findNavController()!!.navigate(R.id.action_settingsFragment_to_accountsFragment)
+        }
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt:52]: Avoid !!

File: WhatsApp/app/src/main/res/layout/fragment_settings.xml [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+    <ImageView
+        android:paddingStart="12dp"
+        android:paddingEnd="12dp"
+        android:id="@+id/ivReferralIcon"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:src="@drawable/ic_contact" />
+
+    <TextView
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/res/layout/fragment_settings.xml:294]: This TextView needs to be aligned center vertically

File: WhatsApp/app/src/main/res/layout/fragment_settings.xml [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_alignParentTop="true"
+            android:layout_centerHorizontal="true"
+            android:src="@drawable/common_full_open_on_phone">
+
+        </ImageView>
+    </androidx.cardview.widget.CardView>
+
+        <TextView
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/res/layout/fragment_settings.xml:48]: Would we be able to have more margins on these views next to the profile picture?

File: WhatsApp/app/src/main/res/layout/fragment_settings.xml [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+            android:id="@+id/userStatus"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="6dp"
+            android:layout_marginTop="50dp"
+            android:layout_toEndOf="@+id/cardView"
+            android:text="Example of status" />
+
+        <ImageView
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/res/layout/fragment_settings.xml:66]: Think this needs a little bit less margin to the right on the QR code

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            val i = Intent(
+                Intent.ACTION_PICK,
+                MediaStore.Images.Media.INTERNAL_CONTENT_URI
+            )
+            val ACTIVITY_SELECT_IMAGE = 1234
+            startActivityForResult(i, ACTIVITY_SELECT_IMAGE)
+        }
+
+        binding.accountTransfer.setOnClickListener {
+            findNavController()!!.navigate(R.id.action_settingsFragment_to_accountsFragment)
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt]: I don't think these null assertions "!!" are needed as they would just be causing warnings in Android Studio

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    override fun initViews() {
+        super.initViews()
+        binding.displayPhoto.setOnClickListener {
+            val i = Intent(
+                Intent.ACTION_PICK,
+                MediaStore.Images.Media.INTERNAL_CONTENT_URI
+            )
+            val ACTIVITY_SELECT_IMAGE = 1234
+            startActivityForResult(i, ACTIVITY_SELECT_IMAGE)
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt:32]: `startActivityForResult` is deprecated in the more recent Java versions and as we're trying to keep this project super modern and practice with newest tech and learn it and from each other, I think we should take this chance to convert over to `ActivityResultLauncher` here is an interesting article I found on it 

https://medium.com/droid-log/androidx-activity-result-apis-the-new-way-7cfc949a803c

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+import com.example.whatsapp.base.BaseFragment
+import com.example.whatsapp.databinding.FragmentSettingsBinding
+import dagger.hilt.android.AndroidEntryPoint
+
+
+@AndroidEntryPoint
+class SettingsFragment : BaseFragment<FragmentSettingsBinding>() {
+
+
+    private lateinit var navController: NavController
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt]: This isn't needed as a variable in a fragment as we can just call `findNavController()`

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt [hunk 8]
```diff
@@ -45,10 +42,17 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>() {
...
 
         binding.fabNewConversation.setOnClickListener {
-            findNavController().navigate(R.id.action_homeFragment_to_contactsFragment)
+            findNavController().navigate(R.id.action_homeFragment_to_chatsFragment)
         }
     }
 
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.itemSettings -> findNavController()!!.navigate(R.id.action_homeFragment_to_settingsFragment)
```
Comment (Josh-Owen) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt]: This non-null assertion is not needed either

Comment (UgurDenli) [reply WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt]: Fixed

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt (hunk 6 above)
Comment (UgurDenli) [reply WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/SettingsFragment.kt:32]: Left for later due to how long it will take

**Review by UgurDenli — COMMENTED** [review]
added changes

**Review by Josh-Owen — APPROVED** [review]

**Review by OskarLasota — COMMENTED** [review]
Did we have to replace FragmentContainerView with fragment in XML files?

**Pull request merged**